### PR TITLE
DPG: Add downsampling in derived data

### PR DESCRIPTION
**Pull request by njacazio**

### CODE REVIEW

**Comment by njacazio** [comment]
@mfaggin @arossi81 enjoy!

**Comment by mfaggin** [comment]
> @mfaggin @arossi81 enjoy!

I enjoyed and I find it ok, green light from my side.

File: DPG/Tasks/qaEventTrack.cxx [hunk 1]
```diff
@@ -126,12 +130,24 @@ struct qaEventTrack {
...
     if (selectGoodEvents && !(isRun3 ? collision.sel8() : collision.sel7())) { // currently only sel8 is defined for run3
       return;
     }
+    if (abs(collision.posZ()) > selectMaxVtxZ) {
+      return;
+    }
+    if (fractionOfSampledEvents < 1.f && (static_cast<float>(rand()) / static_cast<float>(RAND_MAX)) > fractionOfSampledEvents) { // Skip events that are not sampled
+      return;
+    }
+    nTableEventCounter++;
```
Comment (mario-krueger) [inline DPG/Tasks/qaEventTrack.cxx:146]: is this counted twice on purpose?

**Review by mario-krueger — DISMISSED** [review]

**Comment by mfaggin** [comment]
> Error while checking build/O2Physics/o2 for [397c9df](https://github.com/AliceO2Group/O2Physics/commit/397c9df2216f38367dfb17b4ee205e55253775af) at 2022-06-08 13:34:
> 
> ```
> ## sw/BUILD/O2Physics-latest/log
> /sw/SOURCES/O2Physics/841-slc7_x86-64/0/DPG/Tasks/qaEventTrack.cxx:63:179: error: narrowing conversion of '1.0e+7' from 'double' to 'int' [-Wnarrowing]
> ninja: build stopped: subcommand failed.
> ```
> 
> Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/841/397c9df2216f38367dfb17b4ee205e55253775af/build_O2Physics_o2/pretty.html).

@njacazio maybe you need to write `10000000` instead of `10e6`, otherwise that's interpreted as `double` I think

**Review by mario-krueger — APPROVED** [review]

**Pull request merged**